auth.go

```diff
@@ -62,6 +67,42 @@ func (a *Auth) CheckPermission(scope, userID string) error {
 	return ErrNotAllowed
 }
 
+// GetUsers returns a list of userIDs having one or more permission scopes
+func (a *Auth) GetUsers() ([]string, error) {
+	keys, err := a.store.Keys()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load permissions")
+	}
+
+	a.logger.Debug("Retrieving all userIDs")
+
+	var userIDs []string
+	for _, key := range keys {
+		userID, err := a.userFromKey(key)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse userID from key")
+		}
+		userIDs = append(userIDs, userID)
+	}
```

Comment:
Since we are iterating over all keys the bot knows we should expect that also non-permission keys are among them.

I suggest you update your unit test to check this is also handled correctly. A simple implementation would be this:

```go
for _, key := range keys {
	if strings.HasPrefix(key, PermissionKeyPrefix) {
		userID := strings.TrimPrefix(key, PermissionKeyPrefix)
		userIDs = append(userIDs, userID)
	}
}
```

The we would also not need the small `userFromKey` function anymore.

Reply:
It took me a while to understand what you meant, but when I added the test (see diff) I realized it would always fail as soon as there is a non-user key in the store.

I've removed the `userFromKey` function as you suggested and I've added the proposed changes in the function. 

I've improved the test by adding a dummy non-user-permission key and also by checking if the resulting user array length is the same as the input.

I think improvements suggested in #31 will address further concerns we may have around key prefixes

Fixed in https://github.com/go-joe/joe/pull/28/commits/8c7ed27f903d842bee363c8bd2f8e6ed85823a84